.github/scripts/release.py

```diff
@@ -0,0 +1,205 @@
+                            self.vnumber,
+                        )
+                print(line, end="")
+
+        return
+
+    def update_index(self, target):
+        """Update the index page with the new release."""
+
+        # Check whether an update is required
+        with FileInput(target) as fd:
+            match = f'a href="vn{self.vnumber}"'
+            found = False
+
+            for line in fd:
+                if match in line:
+                    found = True
+                    break
+
+        if found:
+            # Stop if version is already present
+            return
+
+        # Add the missing version link
+        with FileInput(target, inplace=True) as fd:
+            for line in fd:
+                if 'href="latest/index.html"' in line:
+                    line += " " * 16
+                    line += (
+                        f'<li><a href="vn{self.vnumber}">vn{self.vnumber}</a></li>\n'
+                    )
+                    self.changed = True
+                    logging.info("Added reference to vn%s to index", self.vnumber)
+                print(line, end="")
+
+        return
+
+    def mkdocs(self):
+        """Run sphinx to build the HTML documentation."""
+
+        dest = f"vn{self.vnumber}"
+
+        if not os.path.exists(dest):
+            # Build the sphinx documentation
+            cmd = [
+                "sphinx-build",
+                "-qb",
+                "html",
+                "user_guide/doc/source",
+                f"vn{self.vnumber}",
+            ]
+            run(cmd, check=True)
+            logging.info("Successfully built the sphinx documentation")
+
+        if os.path.exists(dest):
+            # Check directory exists and create a symlink
+            if os.path.islink("latest") and os.readlink("latest") != dest:
+                os.unlink("latest")
+            if not os.path.exists("latest"):
+                os.symlink(dest, "latest")
+                logging.info("Set latest link to %s", dest)
+
+        return
+
+
+def last_log_message(branch):
+    """Get the log message of the last commit to trunk."""
+
+    cmd = ["git", "log", "--pretty=format:message: %s", "--branches", branch, "-1"]
+
+    with Popen(cmd, stdout=PIPE, stderr=STDOUT, encoding="utf-8") as proc:
+        message = "unknown update"
+
+        for line in proc.stdout:
+            if line.startswith("message: "):
+                message = line.split(": ", 1)[-1]
+                break
+
+    return message
+
+
+def commit_changes(args):
+    """Commit any new change to the trunk."""
+
+    cmd = ["git", "add", "-A"]
+    run(cmd, check=False)
+
+    cmd = ["git", "diff", "--cached", "--quiet"]
+    if run(cmd, check=False) == 0:
+        logging.info("no changes need to be committed")
+
+    message = last_log_message(args.trunk_name)
+    message = f"Docs build for {message}"
+
+    cmd = ["git", "commit", "--no-gpg-sign", "-a", "--quiet", "-m", message]
+
+    if not args.no_commit:
```

Comment:
Why the reverse logic here?

Reply:
No good reason.  It's cleaner to flip the logic and return immediately after the if test.  For good measure, I've also moved the cmd definition after the new return to make the behaviour more obvious.